pkg/intercp/components.go

```diff
@@ -39,31 +43,43 @@ func Setup(rt runtime.Runtime) error {
 		InterCpPort: cfg.Server.Port,
 	}
 
+	pool := rt.InterCPClientPool()
+	go pool.StartCleanup(rt.AppContext(), 10*time.Second)
```

Comment:
seems like you always start clean up after you initialise the pool, so maybe this should happen automatically?

Reply:
Do you mean in `NewPool`? I could move it there although for some reason it feels a bit weird to pass context in constructor and start goroutine. I don't think I have specific reasons, so if you prefer this I can move it to the constructor.

Reply:
I'm just thinking about the usage here, if cleanup is always meant to start right after the pool is created then I see no reason making users call it manually. If you don't think that is the case then this can stay as it is 👍